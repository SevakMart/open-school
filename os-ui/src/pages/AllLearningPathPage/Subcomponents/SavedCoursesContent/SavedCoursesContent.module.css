.mainContainer {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 32px 32px;
  justify-content: none;
}/*# sourceMappingURL=SavedCoursesContent.module.css.map */