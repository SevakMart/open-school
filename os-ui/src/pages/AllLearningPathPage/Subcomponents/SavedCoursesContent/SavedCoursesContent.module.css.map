{"version":3,"sources":["SavedCoursesContent.module.scss","../../../../styles/Mixins/GridMixin.scss","SavedCoursesContent.module.css"],"names":[],"mappings":"AAGA;ECFI,aAAA;EACA,qCDEmB;ECDnB,cAAA;EACA,qBAAA;ACCJ","file":"SavedCoursesContent.module.css"}