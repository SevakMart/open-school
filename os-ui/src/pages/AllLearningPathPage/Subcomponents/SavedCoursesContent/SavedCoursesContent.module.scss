@import "../../../../styles/Mixins/FlexMixins.scss";


.coreContent{
    @include flexStyle(column,none,space-between);
    padding-left:6%;
    padding-right:6%;
    margin-top:3rem;
.mainContainer {
    @include flexStyle(row,center,space-between);
    flex-wrap: wrap;
}
}