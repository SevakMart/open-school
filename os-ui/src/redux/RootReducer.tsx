import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
