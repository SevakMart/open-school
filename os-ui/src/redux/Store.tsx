import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
