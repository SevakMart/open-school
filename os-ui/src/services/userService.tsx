import fetchService from './fetchData';

class UserService {
  readonly basePath: string;

  constructor() {
    this.basePath = 'users';
  }

  async getUserCourses(userId:number, userToken:string, params = {}) {
    const response = await fetchService.get(`${this.basePath}/${userId}/courses/enrolled`, params, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async getSuggestedCourses(userId:number, userToken:string) {
    const response = await fetchService.get(`${this.basePath}/${userId}/courses/suggested`, {}, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async getMentors(params:object = {}, token:string) {
    const response = await fetchService.get(`${this.basePath}/mentors`, params, token);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async savePreferredCategories(userId:number, userToken:string, content:Array<number>) {
    const data = await (await fetchService.post(`${this.basePath}/${userId}/categories`, content, {}, userToken)).json();
    return data;
  }

  async saveUserPreferredCourses(userId:number, courseId:number, userToken:string) {
    const data = await (await fetchService.post(`${this.basePath}/${userId}/courses/saved`, { courseId }, {}, userToken)).json();
    return data;
  }

  async getUserSavedCourses(userId:number, userToken:string, params = {}) {
    const data = await (await fetchService.get(`${this.basePath}/${userId}/courses/saved`, params, userToken)).json();
    return data;
  }

  async deleteUserSavedCourses(userId:number, courseId:number, userToken:string, params = {}) {
    const response = await fetchService.delete(`${this.basePath}/${userId}/courses/${courseId}/saved`, params, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async saveUserMentor(userId:number, mentorId:number, userToken:string, params = {}) {
    const data = await (await fetchService.post(`${this.basePath}/${userId}/mentors/${mentorId}`, null, params, userToken)).json();
    return data;
  }

  async deleteUserSavedMentor(userId:number, mentorId:number, userToken:string, params = {}) {
    const response = await fetchService.delete(`${this.basePath}/${userId}/mentors/${mentorId}/saved`, params, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }
}
export default new UserService();
