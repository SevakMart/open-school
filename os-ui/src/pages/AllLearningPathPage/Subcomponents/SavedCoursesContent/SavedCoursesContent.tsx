import React, { useEffect, useState, useContext } from 'react';
import { useTranslation } from 'react-i18next';
import userService from '../../../../services/userService';
import { SuggestedCourseType } from '../../../../types/SuggestedCourseType';
import LearningPath from '../../../../component/LearningPath/LearningPath';
import { userContext } from '../../../../contexts/Contexts';
import styles from './SavedCoursesContent.module.scss';

type CourseListType=SuggestedCourseType & {id:number, isBookMarked?:boolean}

const SavedCoursesContent = () => {
  const { token, id } = useContext(userContext);
  const [savedCourseList, setSavedCourseList] = useState<CourseListType[]>([]);
  const { t } = useTranslation();
  const { mainContainer, coreContent } = styles;

  const handleCourseDeletion = (courseId:number) => {
    userService.deleteUserSavedCourses(id, courseId, token);
    const index = savedCourseList.findIndex((course) => course.id === courseId);
    const savedCourses = savedCourseList;
    savedCourses.splice(index, 1);
    const bookmarkedCourses = savedCourses.map(
      (course:CourseListType) => ({ ...course, isBookMarked: true }),
    );
    setSavedCourseList([...bookmarkedCourses]);
  };

  useEffect(() => {
    userService.getUserSavedCourses(id, token, { page: 0, size: 100 })
      .then((data) => setSavedCourseList([...data.content.map((
        course:CourseListType,
      ) => ({ ...course, isBookMarked: true }))]));
  }, []);

  return (
    <>
      <div className={coreContent}>
        <div className={mainContainer}>
          {savedCourseList.length ? savedCourseList.map((course) => (
            <React.Fragment key={course.title}>
              <LearningPath
                courseInfo={course}
                saveCourse={(courseId:number) => {
                  userService.saveUserPreferredCourses(id, courseId, token);
                }}
                deleteCourse={handleCourseDeletion}
              />
            </React.Fragment>
          )) : <h2 data-testid="Empty data Message">{t('messages.noData.default')}</h2>}
        </div>
      </div>
    </>
  );
};
export default SavedCoursesContent;
