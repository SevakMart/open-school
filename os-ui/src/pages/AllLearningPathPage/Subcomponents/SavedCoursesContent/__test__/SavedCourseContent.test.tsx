import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import SavedCoursesContent from '../SavedCoursesContent';
import userService from '../../../../../services/userService';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: () => 'No data to display',
  }),
}));

const data = {
  content: [
    {
      title: 'JavaScript',
      rating: 5.5,
      difficulty: 'Medium',
      keywords: ['Testing', 'database', 'memory'],
    },
    {
      title: 'React',
      rating: 5.6,
      difficulty: 'Medium',
      keywords: ['Testing', 'database', 'memory'],
    },
  ],
};

const mockUseNavigate = jest.fn();
const mockUseLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => mockUseLocation,
}));

describe('Create test cases for SavedCourseContent component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SavedCoursesContent />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Test if component renders as expected when we do not have saved courses', async () => {
    expect.hasAssertions();
    jest.spyOn(userService, 'getUserSavedCourses').mockResolvedValue({ content: [] });
    render(
      <Provider store={store}>
        <SavedCoursesContent />
      </Provider>,
    );
    const emptyDataheaderElement = await screen.findByTestId('Empty data Message');
    expect(emptyDataheaderElement).toBeInTheDocument();
    expect(emptyDataheaderElement).toHaveTextContent('No data to display');
  });
  test('Test if component renders as expected when we have saved courses', async () => {
    expect.hasAssertions();
    jest.spyOn(userService, 'getUserSavedCourses').mockResolvedValue(data);
    render(
      <Provider store={store}>
        <SavedCoursesContent />
      </Provider>,
    );
    const emptyDataheaderElement = await screen.findByTestId('Empty data Message');
    expect(emptyDataheaderElement).not.toBeInTheDocument();
  });
});
