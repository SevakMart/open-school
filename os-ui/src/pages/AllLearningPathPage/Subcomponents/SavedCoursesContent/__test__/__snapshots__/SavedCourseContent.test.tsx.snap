// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for SavedCourseContent component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="coreContent"
  >
    <div
      class="mainContainer"
    >
      <h2
        data-testid="Empty data Message"
      >
        No data to display :(
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
